### PR TITLE
Delete the fetch shim

### PR DESCRIPTION
**Pull request by steveluscher**
# Summary

Browsers support `fetch`. https://caniuse.com/fetch

Node 18+ supports `fetch`.

React Native supports `fetch`.

Farewell to our trusty `fetch` shim.

# Test Plan

You can still use `undici` Dispatchers in Node: benchmarks still run, but now against `rpc-transport-http` directly.

```shell
cd packages/rpc-transport-http
pnpm benchmark https://api.devnet.solana.com/API_KEY
```

### CODE REVIEW

**Comment by steveluscher** [comment]
* **#2267** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2267?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @steveluscher and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/rpc-transport-http/package.json [hunk 1]
```diff
@@ -87,9 +87,11 @@
         "jest-runner-eslint": "^2.2.0",
         "jest-runner-prettier": "^1.0.0",
         "prettier": "^3.1",
+        "tinybench": "^2.6.0",
         "tsup": "^8.0.1",
         "typescript": "^5.2.2",
-        "version-from-git": "^1.1.1"
+        "version-from-git": "^1.1.1",
+        "zx": "^7.2.3"
```
Comment (lorisleiva) [inline packages/rpc-transport-http/package.json:94]: One of us! 🙌

**Review by lorisleiva — APPROVED** [review]

**Comment by steveluscher** [comment]
For completeness, this had the side-effect of making the library work on Stackblitz again: https://stackblitz.com/edit/stackblitz-starters-pgmlsb?file=index.ts

See also https://github.com/stackblitz/webcontainer-core/issues/1363

**Comment by steveluscher** [comment]
Even better now on CodeSandbox https://codesandbox.io/p/devbox/distracted-nobel-l3z3nh?file=%2Findex.ts%3A71%2C72

**Pull request merged**